test/gtest/ucp/test_ucp_proto.cc

```diff
@@ -353,3 +358,170 @@ UCS_TEST_P(test_perf_node, replace_null)
 }
 
 UCP_INSTANTIATE_TEST_CASE_TLS(test_perf_node, all, "all")
+
+
+class test_rndv_abort : public test_ucp_proto {
+public:
+    virtual ucp_ep_params_t get_ep_params()
+    {
+        ucp_ep_params_t ep_params = ucp_test::get_ep_params();
+        ep_params.field_mask     |= UCP_EP_PARAM_FIELD_ERR_HANDLING_MODE |
+                                    UCP_EP_PARAM_FIELD_ERR_HANDLER;
+        ep_params.err_mode        = UCP_ERR_HANDLING_MODE_PEER;
+        ep_params.err_handler.cb  = err_handler_cb;
+        ep_params.err_handler.arg = this;
+        return ep_params;
+    }
+
+    void init()
+    {
+        self = this;
+
+        test_ucp_proto::init();
+
+        receiver().connect(&sender(), get_ep_params());
+    }
+
+    void protos_replace_progress()
+    {
+        for (int proto_id = 0; proto_id < ucp_protocols_count(); ++proto_id) {
+            auto proto = const_cast<ucp_proto_t*>(ucp_protocols[proto_id]);
+            m_progress_storage.emplace(proto, progress_array{});
+
+            for (int stage = 0; stage < UCP_PROTO_STAGE_LAST; ++stage) {
+                m_progress_storage[proto][stage] = proto->progress[stage];
+                proto->progress[stage]           = &progress_wrapper;
+            }
+        }
+    }
+
+    void protos_restore_progress()
+    {
+        for (int proto_id = 0; proto_id < ucp_protocols_count(); ++proto_id) {
+            auto proto = const_cast<ucp_proto_t*>(ucp_protocols[proto_id]);
+
+            for (int stage = 0; stage < UCP_PROTO_STAGE_LAST; ++stage) {
+                proto->progress[stage] = m_progress_storage[proto][stage];
+            }
+        }
+    }
+
+    void replace_ep_put_zcopy(ucp_ep_h ep) {
+        for (auto lane = 0; lane < ucp_ep_num_lanes(ep); ++lane) {
+            if (lane == ucp_ep_get_cm_lane(ep)) {
+                continue;
+            }
+
+            auto iface  = ucp_ep_get_lane(ep, lane)->iface;
+            auto result = m_sender_uct_ops.emplace(iface, iface->ops);
+            if (result.second) {
+                iface->ops.ep_put_zcopy =
+                        reinterpret_cast<uct_ep_put_zcopy_func_t>(
+                                ucs_empty_function_return_unsupported);
+            }
+        }
+    }
+
+    static ucs_status_t progress_wrapper(uct_pending_req_t *uct_req)
+    {
+        ucs_status_t status;
+        auto req   = ucs_container_of(uct_req, ucp_request_t, send.uct);
+        auto proto = req->send.proto_config->proto;
+        auto stage = req->send.proto_stage;
+        std::string proto_name = req->send.proto_config->proto->name;
+
+        if (proto_name == "rndv/put/zcopy") {
```

Comment:
maybe like this to unify?
```suggestion
        auto proto_name = req->send.proto_config->proto->name;

        if (std::string(proto_name) == "rndv/put/zcopy") {
```